code/game/objects/items/devices/scanners.dm

```diff
@@ -437,7 +440,6 @@ REAGENT SCANNER
 			M.put_in_hands(P)
 			to_chat(M, "<span class='notice'>Report printed. Log cleared.<span>")
 			datatoprint = ""
-			scanning = 1
 	else
 		to_chat(usr, "<span class='notice'>[src] has no logs or is already in use.</span>")
 

```

Comment:
I think there should be a `scanning = FALSE` here

Reply:
Not necessary, the else statement here is for the `if(!scanning)` check at the beginning to prevent you from printing when the device is still scanning something.

I would however recommend changing the `if(!scanning)` to `if(!scanning || !datatoprint)` so it checks if there is any data to print and not just if the device is currently in use

Reply:
Oh, right. I put this in the wrong place then -- line 440 should have `scanning = FALSE`, so it's not always set to `TRUE` (instead `scanning` should be set to `FALSE` after the report is printed).

Reply:
actually, you want to check for datatoprint having a value, I mistyped that in my original comment but have since edited it to fix that mistake.
You are correct about needing the scanning = FALSE after clearing the datatoprint var though